src/vs/workbench/parts/terminal/electron-browser/terminalProcessManager.ts

```diff
@@ -101,25 +101,31 @@ export class TerminalProcessManager implements ITerminalProcessManager {
 			const activeWorkspaceRootUri = this._historyService.getLastActiveWorkspaceRoot(Schemas.file);
 			this.initialCwd = terminalEnvironment.getCwd(shellLaunchConfig, activeWorkspaceRootUri, this._configHelper.config.cwd);
 
-			// Resolve env vars from config and shell
-			const lastActiveWorkspaceRoot = this._workspaceContextService.getWorkspaceFolder(activeWorkspaceRootUri);
-			const platformKey = platform.isWindows ? 'windows' : (platform.isMacintosh ? 'osx' : 'linux');
-			const envFromConfig = terminalEnvironment.resolveConfigurationVariables(this._configurationResolverService, { ...this._configHelper.config.env[platformKey] }, lastActiveWorkspaceRoot);
-			const envFromShell = terminalEnvironment.resolveConfigurationVariables(this._configurationResolverService, { ...shellLaunchConfig.env }, lastActiveWorkspaceRoot);
-			shellLaunchConfig.env = envFromShell;
-
-			// Merge process env with the env from config and from shellLaunchConfig
-			const env = { ...process.env };
-			terminalEnvironment.mergeEnvironments(env, envFromConfig);
-			terminalEnvironment.mergeEnvironments(env, shellLaunchConfig.env);
-
-			// Sanitize the environment, removing any undesirable VS Code and Electron environment
-			// variables
-			terminalEnvironment.sanitizeEnvironment(env);
-
-			// Adding other env keys necessary to create the process
-			const locale = this._configHelper.config.setLocaleVariables ? platform.locale : undefined;
-			terminalEnvironment.addTerminalEnvironmentKeys(env, locale);
+			const env = {};
+			if (shellLaunchConfig.noInheritEnv) {
+				terminalEnvironment.mergeEnvironments(env, shellLaunchConfig.env);
```

Comment:
Do we want the `envFromConfig` also included here? That's pulled from user and workspace `terminal.integrated.end.windows` setting?